services/collectd/service.go

```diff
@@ -385,6 +390,53 @@ func (s *Service) writePoints() {
 	}
 }
 
+// UnmarshalValueListPacked is an alternative to the original UnmarshalValueList
+// The difference is that the original provided measurements like (PLUGIN_DSNAME, ["value",xxx])
+// while this one will provide measurements like (PLUGIN, {["DSNAME",xxx]})
+// this effectively joins collectd data that should go together, such as:
+//  (df, {["used",1000],["free",2500]})
+func (s *Service) UnmarshalValueListPacked(vl *api.ValueList) []models.Point {
+	timestamp := vl.Time.UTC()
+
+	var name = vl.Identifier.Plugin
+	tags := make(map[string]string)
+	fields := make(map[string]interface{})
+
+	if vl.Identifier.Host != "" {
+		tags["host"] = vl.Identifier.Host
+	}
+	if vl.Identifier.PluginInstance != "" {
+		tags["instance"] = vl.Identifier.PluginInstance
+	}
+	if vl.Identifier.Type != "" {
+		tags["type"] = vl.Identifier.Type
+	}
+	if vl.Identifier.TypeInstance != "" {
+		tags["type_instance"] = vl.Identifier.TypeInstance
+	}
+
+	for i := range vl.Values {
+		fieldName := vl.DSName(i)
+		switch value := vl.Values[i].(type) {
```

Comment:
It's generally preferable to use the two-argument range function rather than indexing again, i.e.

```
for i, v := range vl.Values {
  fieldName := vl.DSName(i)
  switch val := v.(type) {
```